### ISSUE TITLE
Signal Received not working with Mongodb UseMongoDbPersistence

### ISSUE THREAD

**Issue by serenabenny** [elsa-workflows/elsa-core#1110]
Version: 2.1.0-preview.257
I have tested this issue with simple document approval workflow examples.
In the code below if I use .UseEntityFrameworkPersistence(ef => ef.UseSqlite()) everything runs as expected.

```
{
  "$id": "1",
  "definitionId": "f18e318f98be405da00e1f43f34c3e40",
  "versionId": "9b6a5272573d4cc2b8957278bca2791b",
  "name": "VisualDocumentApprovalWorkflow",
  "displayName": "Visual Document Approval Workflow",
  "version": 1,
  "variables": {
    "$id": "2",
    "data": {}
  },
  "customAttributes": {
    "$id": "3",
    "data": {}
  },
  "isSingleton": false,
  "persistenceBehavior": "WorkflowBurst",
  "deleteCompletedInstances": false,
  "isPublished": true,
  "isLatest": true,
  "activities": [
    {
      "$id": "4",
      "activityId": "6923b676-1a0d-4fcb-8000-7a18dd571882",
      "type": "HttpEndpoint",
      "displayName": "HTTP Endpoint",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "5",
          "name": "Path",
          "expressions": {
            "$id": "6",
            "Literal": "/v2/documents"
          }
        },
        {
          "$id": "7",
          "name": "Methods",
          "expressions": {
            "$id": "8",
            "Json": "[\"POST\"]"
          }
        },
        {
          "$id": "9",
          "name": "ReadContent",
          "expressions": {
            "$id": "10",
            "Literal": "true"
          }
        },
        {
          "$id": "11",
          "name": "TargetType",
          "expressions": {
            "$id": "12"
          }
        }
      ]
    },
    {
      "$id": "13",
      "activityId": "a5b29101-244b-410d-a64f-84bde0052e37",
      "type": "SetVariable",
      "displayName": "Set Variable",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "14",
          "name": "VariableName",
          "expressions": {
            "$id": "15",
            "Literal": "Document"
          }
        },
        {
          "$id": "16",
          "name": "Value",
          "syntax": "JavaScript",
          "expressions": {
            "$id": "17",
            "JavaScript": "input.Body"
          }
        }
      ]
    },
    {
      "$id": "18",
      "activityId": "496dd4a7-92cb-4e0b-87c2-453e9f54ea62",
      "type": "SendEmail",
      "displayName": "Send Email",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "19",
          "name": "From",
          "expressions": {
            "$id": "20",
            "Literal": "system@amce.com"
          }
        },
        {
          "$id": "21",
          "name": "To",
          "expressions": {
            "$id": "22",
            "Json": "[\"benny.lin@amce.com\"]"
          }
        },
        {
          "$id": "23",
          "name": "Subject",
          "syntax": "Liquid",
          "expressions": {
            "$id": "24",
            "Literal": "{{ Variables.Document.Author.Name }} is waiting for your review!",
            "Liquid": "Document received from {{Variables.Document.Author.Name}}"
          }
        },
        {
          "$id": "25",
          "name": "Body",
          "syntax": "Liquid",
          "expressions": {
            "$id": "26",
            "Liquid": "Document from {{ Variables.Document.Author.Name }} received for review.<br><a href=\"{{ \"Approve\" | signal_url }}\">Approve</a> or <a href=\"{{ \"Reject\" | signal_url }}\">Reject</a>"
          }
        },
        {
          "$id": "27",
          "name": "Cc",
          "expressions": {
            "$id": "28"
          }
        },
        {
          "$id": "29",
          "name": "Bcc",
          "expressions": {
            "$id": "30"
          }
        }
      ]
    },
    {
      "$id": "31",
      "activityId": "dcb4e595-c297-49c9-b081-2d0d424d8ef5",
      "type": "WriteHttpResponse",
      "displayName": "HTTP Response",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "32",
          "name": "Content",
          "expressions": {
            "$id": "33",
            "Literal": "<h1>Request for Approval Sent</h1><p>Your document has been received and will be reviewed shortly.</p>"
          }
        },
        {
          "$id": "34",
          "name": "ContentType",
          "expressions": {
            "$id": "35",
            "Literal": "text/html"
          }
        },
        {
          "$id": "36",
          "name": "StatusCode",
          "expressions": {
            "$id": "37"
          }
        },
        {
          "$id": "38",
          "name": "CharSet",
          "expressions": {
            "$id": "39"
          }
        },
        {
          "$id": "40",
          "name": "ResponseHeaders",
          "expressions": {
            "$id": "41"
          }
        }
      ]
    },
    {
      "$id": "42",
      "activityId": "f5550de6-78d7-42c1-abc3-851c1602dd8a",
      "type": "Fork",
      "displayName": "Fork",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "43",
          "name": "Branches",
          "expressions": {
            "$id": "44",
            "Json": "[\"Approve\",\"Reject\",\"Remind\"]"
          }
        }
      ]
    },
    {
      "$id": "45",
      "activityId": "7dff843a-4151-47b5-a2c4-104ea5f7a57a",
      "type": "SignalReceived",
      "displayName": "Signal Received",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "46",
          "name": "Signal",
          "expressions": {
            "$id": "47",
            "Literal": "Approve"
          }
        },
        {
          "$id": "48",
          "name": "Scope",
          "expressions": {
            "$id": "49"
          }
        }
      ]
    },
    {
      "$id": "50",
      "activityId": "264cec59-7cea-4064-b460-aed898d54b3e",
      "type": "SendEmail",
      "displayName": "Send Email",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "51",
          "name": "From",
          "expressions": {
            "$id": "52",
            "Literal": "system@amce.com"
          }
        },
        {
          "$id": "53",
          "name": "To",
          "syntax": "JavaScript",
          "expressions": {
            "$id": "54",
            "JavaScript": "[Document.Author.Email]"
          }
        },
        {
          "$id": "55",
          "name": "Subject",
          "syntax": "Liquid",
          "expressions": {
            "$id": "56",
            "Liquid": "Document {{ Variables.Document.Id }} Approved!"
          }
        },
        {
          "$id": "57",
          "name": "Body",
          "syntax": "Liquid",
          "expressions": {
            "$id": "58",
            "Liquid": "Great job {{ Variables.Document.Author.Name }}, that document is perfect."
          }
        },
        {
          "$id": "59",
          "name": "Cc",
          "expressions": {
            "$id": "60"
          }
        },
        {
          "$id": "61",
          "name": "Bcc",
          "expressions": {
            "$id": "62"
          }
        }
      ]
    },
    {
      "$id": "63",
      "activityId": "632ac28a-7aae-499e-8166-cff321603a6b",
      "type": "SignalReceived",
      "displayName": "Signal Received",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "64",
          "name": "Signal",
          "expressions": {
            "$id": "65",
            "Literal": "Reject"
          }
        },
        {
          "$id": "66",
          "name": "Scope",
          "expressions": {
            "$id": "67"
          }
        }
      ]
    },
    {
      "$id": "68",
      "activityId": "e8079d3e-4da2-4f9d-8f6c-610ee50f937f",
      "type": "SendEmail",
      "displayName": "Send Email",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "69",
          "name": "From",
          "expressions": {
            "$id": "70",
            "Literal": "system@amce.com"
          }
        },
        {
          "$id": "71",
          "name": "To",
          "syntax": "JavaScript",
          "expressions": {
            "$id": "72",
            "JavaScript": "[Document.Author.Email]"
          }
        },
        {
          "$id": "73",
          "name": "Subject",
          "syntax": "Liquid",
          "expressions": {
            "$id": "74",
            "Liquid": "Document {{ Variables.Document.Id }} Rejected"
          }
        },
        {
          "$id": "75",
          "name": "Body",
          "syntax": "Liquid",
          "expressions": {
            "$id": "76",
            "Liquid": "Nice try {{ Variables.Document.Author.Name }}, but that document needs work."
          }
        },
        {
          "$id": "77",
          "name": "Cc",
          "expressions": {
            "$id": "78"
          }
        },
        {
          "$id": "79",
          "name": "Bcc",
          "expressions": {
            "$id": "80"
          }
        }
      ]
    },
    {
      "$id": "81",
      "activityId": "e70f6d8d-8954-4446-b0b9-fced2b59e599",
      "type": "Timer",
      "displayName": "Timer",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "82",
          "name": "Timeout",
          "syntax": "JavaScript",
          "expressions": {
            "$id": "83",
            "JavaScript": "Duration.FromDays(1)"
          }
        }
      ]
    },
    {
      "$id": "84",
      "activityId": "cb5b4a05-e5a4-4244-ab22-9780e375effc",
      "type": "SendEmail",
      "displayName": "Send Email",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "85",
          "name": "From",
          "expressions": {
            "$id": "86",
            "Literal": "system@amce.com"
          }
        },
        {
          "$id": "87",
          "name": "To",
          "syntax": "JavaScript",
          "expressions": {
            "$id": "88",
            "JavaScript": "[Document.Approver.Email]"
          }
        },
        {
          "$id": "89",
          "name": "Subject",
          "syntax": "Liquid",
          "expressions": {
            "$id": "90",
            "Liquid": "{{ Variables.Document.Author.Name }} is waiting for your review!"
          }
        },
        {
          "$id": "91",
          "name": "Body",
          "syntax": "Liquid",
          "expressions": {
            "$id": "92",
            "Liquid": "Don't forget to review document {{ Variables.Document.Id }}.<br><a href=\"{{ \"Approve\" | signal_url }}\">Approve</a> or <a href=\"{{ \"Reject\" | signal_url }}\">Reject</a>"
          }
        },
        {
          "$id": "93",
          "name": "Cc",
          "expressions": {
            "$id": "94"
          }
        },
        {
          "$id": "95",
          "name": "Bcc",
          "expressions": {
            "$id": "96"
          }
        }
      ]
    },
    {
      "$id": "97",
      "activityId": "ada4ad24-7cfd-4b07-a2f7-36ed3b2f0474",
      "type": "Join",
      "displayName": "Join",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "98",
          "name": "Mode",
          "expressions": {
            "$id": "99",
            "Literal": "WaitAny"
          }
        }
      ]
    },
    {
      "$id": "100",
      "activityId": "5eff29d0-c501-4f38-b085-46200d302472",
      "type": "WriteHttpResponse",
      "displayName": "HTTP Response",
      "persistWorkflow": false,
      "loadWorkflowContext": false,
      "saveWorkflowContext": false,
      "persistOutput": false,
      "properties": [
        {
          "$id": "101",
          "name": "Content",
          "expressions": {
            "$id": "102",
            "Literal": "Thanks for the hard work!"
          }
        },
        {
          "$id": "103",
          "name": "ContentType",
          "expressions": {
            "$id": "104",
            "Literal": "text/html"
          }
        },
        {
          "$id": "105",
          "name": "StatusCode",
          "expressions": {
            "$id": "106"
          }
        },
        {
          "$id": "107",
          "name": "CharSet",
          "expressions": {
            "$id": "108"
          }
        },
        {
          "$id": "109",
          "name": "ResponseHeaders",
          "expressions": {
            "$id": "110"
          }
        }
      ]
    }
  ],
  "connections": [
    {
      "$id": "111",
      "sourceActivityId": "6923b676-1a0d-4fcb-8000-7a18dd571882",
      "targetActivityId": "a5b29101-244b-410d-a64f-84bde0052e37",
      "outcome": "Done"
    },
    {
      "$id": "112",
      "sourceActivityId": "a5b29101-244b-410d-a64f-84bde0052e37",
      "targetActivityId": "496dd4a7-92cb-4e0b-87c2-453e9f54ea62",
      "outcome": "Done"
    },
    {
      "$id": "113",
      "sourceActivityId": "496dd4a7-92cb-4e0b-87c2-453e9f54ea62",
      "targetActivityId": "dcb4e595-c297-49c9-b081-2d0d424d8ef5",
      "outcome": "Done"
    },
    {
      "$id": "114",
      "sourceActivityId": "dcb4e595-c297-49c9-b081-2d0d424d8ef5",
      "targetActivityId": "f5550de6-78d7-42c1-abc3-851c1602dd8a",
      "outcome": "Done"
    },
    {
      "$id": "115",
      "sourceActivityId": "f5550de6-78d7-42c1-abc3-851c1602dd8a",
      "targetActivityId": "7dff843a-4151-47b5-a2c4-104ea5f7a57a",
      "outcome": "Approve"
    },
    {
      "$id": "116",
      "sourceActivityId": "7dff843a-4151-47b5-a2c4-104ea5f7a57a",
      "targetActivityId": "264cec59-7cea-4064-b460-aed898d54b3e",
      "outcome": "Done"
    },
    {
      "$id": "117",
      "sourceActivityId": "f5550de6-78d7-42c1-abc3-851c1602dd8a",
      "targetActivityId": "632ac28a-7aae-499e-8166-cff321603a6b",
      "outcome": "Reject"
    },
    {
      "$id": "118",
      "sourceActivityId": "632ac28a-7aae-499e-8166-cff321603a6b",
      "targetActivityId": "e8079d3e-4da2-4f9d-8f6c-610ee50f937f",
      "outcome": "Done"
    },
    {
      "$id": "119",
      "sourceActivityId": "f5550de6-78d7-42c1-abc3-851c1602dd8a",
      "targetActivityId": "e70f6d8d-8954-4446-b0b9-fced2b59e599",
      "outcome": "Remind"
    },
    {
      "$id": "120",
      "sourceActivityId": "e70f6d8d-8954-4446-b0b9-fced2b59e599",
      "targetActivityId": "cb5b4a05-e5a4-4244-ab22-9780e375effc",
      "outcome": "Done"
    },
    {
      "$id": "121",
      "sourceActivityId": "cb5b4a05-e5a4-4244-ab22-9780e375effc",
      "targetActivityId": "e70f6d8d-8954-4446-b0b9-fced2b59e599",
      "outcome": "Done"
    },
    {
      "$id": "122",
      "sourceActivityId": "264cec59-7cea-4064-b460-aed898d54b3e",
      "targetActivityId": "ada4ad24-7cfd-4b07-a2f7-36ed3b2f0474",
      "outcome": "Done"
    },
    {
      "$id": "123",
      "sourceActivityId": "e8079d3e-4da2-4f9d-8f6c-610ee50f937f",
      "targetActivityId": "ada4ad24-7cfd-4b07-a2f7-36ed3b2f0474",
      "outcome": "Done"
    },
    {
      "$id": "124",
      "sourceActivityId": "ada4ad24-7cfd-4b07-a2f7-36ed3b2f0474",
      "targetActivityId": "5eff29d0-c501-4f38-b085-46200d302472",
      "outcome": "Done"
    }
  ],
  "id": "9b6a5272573d4cc2b8957278bca2791b"
}
```

**Comment by serenabenny** [comment]
The exception message.
```
An unhandled exception occurred while processing the request.
InvalidCastException: Invalid cast from 'System.Int64' to 'Elsa.Activities.Signaling.SignalScope'.
System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)


InvalidCastException: Invalid cast from 'System.Int64' to 'Elsa.Activities.Signaling.SignalScope'.
System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
long.System.IConvertible.ToType(Type type, IFormatProvider provider)
Elsa.ObjectConverter.ConvertTo(object value, Type targetType)
Elsa.JObjectExtensions.GetState(IDictionary<string, object> state, string key, Type targetType)
Elsa.Services.Models.ActivityExecutionContext.GetState(string propertyName, Type targetType)
Elsa.Services.ActivityActivator.ApplyStoredValues(ActivityExecutionContext context, IActivity activity)
Elsa.Services.ActivityActivator.ActivateActivityAsync(ActivityExecutionContext context, Type type, CancellationToken cancellationToken)
Elsa.ActivityTypeProviders.TypeBasedActivityProvider+<>c__DisplayClass6_0+<<CreateActivityTypeAsync>b__0>d.MoveNext()
System.Threading.Tasks.ValueTask<TResult>.get_Result()
Elsa.Services.WorkflowRunner.CanExecuteAsync(WorkflowExecutionContext workflowExecutionContext, IActivityBlueprint activityBlueprint, object input, bool resuming, CancellationToken cancellationToken)
System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable<TResult>+ConfiguredValueTaskAwaiter.GetResult()
Elsa.Services.WorkflowRunner.ResumeWorkflowAsync(WorkflowExecutionContext workflowExecutionContext, IActivityBlueprint activityBlueprint, object input, CancellationToken cancellationToken)
Elsa.Services.WorkflowRunner.RunWorkflowAsync(IWorkflowBlueprint workflowBlueprint, WorkflowInstance workflowInstance, string activityId, object input, CancellationToken cancellationToken)
Elsa.Services.WorkflowResumer.ResumeWorkflowAsync(WorkflowInstance workflowInstance, string activityId, object input, CancellationToken cancellationToken)
Elsa.Services.WorkflowInstanceExecutor.ExecuteAsync(WorkflowInstance workflowInstance, string activityId, object input, CancellationToken cancellationToken)
Elsa.Decorators.LockingWorkflowInstanceExecutor.ExecuteAsync(string workflowInstanceId, string activityId, object input, CancellationToken cancellationToken)
Elsa.Decorators.LockingWorkflowInstanceExecutor.ExecuteAsync(string workflowInstanceId, string activityId, object input, CancellationToken cancellationToken)
Elsa.Decorators.LockingWorkflowInstanceExecutor.ExecuteAsync(string workflowInstanceId, string activityId, object input, CancellationToken cancellationToken)
Elsa.Services.WorkflowLaunchpad.ExecutePendingWorkflowsAsync(IEnumerable<PendingWorkflow> pendingWorkflows, object input, CancellationToken cancellationToken)
Elsa.Services.WorkflowLaunchpad.CollectAndExecuteWorkflowsAsync(CollectWorkflowsContext context, object input, CancellationToken cancellationToken)
Elsa.Activities.Signaling.Services.Signaler.TriggerSignalAsync(string signal, object input, string workflowInstanceId, string correlationId, CancellationToken cancellationToken)
Elsa.Activities.Signaling.Services.Signaler.TriggerSignalTokenAsync(string token, object input, CancellationToken cancellationToken)
Open.Linq.AsyncExtensions.Extensions.ToList<TSource>(Task<IEnumerable<TSource>> source)
Elsa.Activities.Http.Endpoints.Signals.TriggerEndpoint.Handle(string token)
Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor+TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, object controller, object[] arguments)
System.Threading.Tasks.ValueTask<TResult>.get_Result()
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
Elsa.Activities.Http.Middleware.HttpEndpointMiddleware.InvokeAsync(HttpContext httpContext, IWorkflowLaunchpad workflowLaunchpad, IWorkflowInstanceStore workflowInstanceStore, IWorkflowRegistry workflowRegistry, IWorkflowBlueprintReflector workflowBlueprintReflector, IEnumerable<IHttpRequestBodyParser> contentParsers)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by sfmskywalker** [comment]
Might have the same root cause as #1088.

**Comment by sfmskywalker** [comment]
Fixed via 4f9f56fd976a538b0ad9d3e16aab620d8126e357
